File: python/examples/extract_document.py
"""Extract Markdown from a public document URL."""

import os
import sys

from gettxt import GettxtClient


def main() -> None:
    if len(sys.argv) != 2:
        raise SystemExit("Usage: extract_document.py <public-document-url>")

    api_key = os.environ.get("GETTXT_API_KEY")
    if not api_key:
        raise SystemExit("Set GETTXT_API_KEY before running this example")

    result = GettxtClient(api_key=api_key).extract(
        [sys.argv[1]],
        output_format="markdown",
    )
    print(result.all_text)


if __name__ == "__main__":
    main()

File: typescript/examples/extract-document.ts
import { GettxtClient } from "../src/index.js";

const apiKey = process.env.GETTXT_API_KEY;
const documentUrl = process.argv[2];

if (!apiKey) {
  throw new Error("Set GETTXT_API_KEY before running this example");
}
if (!documentUrl) {
  throw new Error("Usage: extract-document <public-document-url>");
}

const client = new GettxtClient({ apiKey });
const result = await client.extract({
  documentUris: [documentUrl],
  outputFormat: "markdown",
});

console.log(result.all_text);

